h/models/organization.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+
+
+import sqlalchemy as sa
+import slugify
+
+from h.db import Base
+from h.db import mixins
+
+
+ORGANIZATION_NAME_MIN_CHARS = 1
+ORGANIZATION_NAME_MAX_CHARS = 25
+ORGANIZATION_LOGO_MAX_CHARS = 10000
+
+
+class Organization(Base, mixins.Timestamps):
+    __tablename__ = 'organization'
+
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+
+    name = sa.Column(sa.UnicodeText(), nullable=False, index=True)
+
+    logo = sa.Column(sa.UnicodeText())
+
+    def __init__(self, **kwargs):
+        super(Organization, self).__init__(**kwargs)
+
+    @property
+    def slug(self):
+        """A version of this organization's name suitable for use in a URL."""
+        return slugify.slugify(self.name)
+
+    @sa.orm.validates('name')
+    def validate_name(self, key, name):
+        if not ORGANIZATION_NAME_MIN_CHARS <= len(name) <= ORGANIZATION_NAME_MAX_CHARS:
+            raise ValueError(
+                'name must be between {min} and {max} characters long'
+                .format(min=ORGANIZATION_NAME_MIN_CHARS,
+                        max=ORGANIZATION_NAME_MAX_CHARS))
+        return name
+
+    @sa.orm.validates('logo')
+    def validate_logo(self, key, logo):
+        if not len(logo) <= ORGANIZATION_LOGO_MAX_CHARS:
```

Comment:
If we are going to stipulate that the logo is an SVG then I think it would be a good idea to validate that the content is at least valid XML with an SVG root tag. Something like:

```py
from xml.etree import ElementTree
...
root = ElementTree.fromstring(logo) # raises if not well-formed
if root.tag != '{http://www.w3.org/2000/svg}svg'
  raise ... # raise if not an SVG document
```

Reply:
That sounds good.